anaconda_project/internal/cli/run.py

```diff
@@ -37,29 +38,36 @@ def run_command(project_dir, ui_mode, conda_environment, command_name, extra_com
         Does not return if successful.
     """
     project = load_project(project_dir)
-    environ = None
 
-    command = _command_from_name(project, command_name)
+    if project.has_bootstrap_env_spec and not project.is_running_in_bootstrap_env:
+        print("Project should be ran by bootstrap env... fixing.")
+        create_bootstrap_env(project)
+        run_on_bootstrap_env(project)
+    else:
+        environ = None
+        command = _command_from_name(project, command_name)
 
-    result = prepare_with_ui_mode_printing_errors(project,
-                                                  ui_mode=ui_mode,
-                                                  env_spec_name=conda_environment,
-                                                  command=command,
-                                                  extra_command_args=extra_command_args,
-                                                  environ=environ)
+        result = prepare_with_ui_mode_printing_errors(project,
+                                                      ui_mode=ui_mode,
+                                                      env_spec_name=conda_environment,
+                                                      command=command,
+                                                      extra_command_args=extra_command_args,
+                                                      environ=environ)
 
-    if result.failed:
-        # errors were printed already
-        return
-    elif result.command_exec_info is None:
-        print("No known run command for project %s; try adding a 'commands:' section to anaconda-project.yml" %
-              project_dir,
-              file=sys.stderr)
-    else:
-        try:
-            result.command_exec_info.execvpe()
-        except OSError as e:
-            print("Failed to execute '%s': %s" % (" ".join(result.command_exec_info.args), e.strerror), file=sys.stderr)
+        if result.failed:
+            # errors were printed already
+            return
```

Comment:
Why do we need to return? I feel having returns in the middle of a routine/procedure/method/func is weird. This is more like a procedure than a function, so it does not return anything, it just does stuff. In this case, a `pass` feels more correct, besides if if entered the if, there is no code that "might get executed" after the `else`

Reply:
This code is not part of the PR (it has just been moved, so the `return` stmnt was already there). Returning in the middle of function-like object does not bother me.. I know returning the result of a function at the end of a function (forcing this "standard" pattern maybe storing results in a variable that you carry until the end) is what have been thought for decades but I think it is ok to return prematuraly. Especially in Python and if this kind of helps understand the overall function behavior.

I'd be -0 about changing it.